### ISSUE TITLE
Phi 3.5 MoE Instruct ISQ: `Forward step expected a PagedAttention input metadata.`

### ISSUE THREAD

**Issue by Remember20240719** [EricLBuehler/mistral.rs#757]
## Describe the bug

Hello, I tried the command you provided in https://github.com/EricLBuehler/mistral.rs/issues/743#issuecomment-2325421505: 

```
export RUST_BACKTRACE=1
./target/release/./mistralrs-server --port 1234 --isq Q6K -i plain -m $MODELS/microsoft/Phi-3.5-MoE-instruct -a phi3.5moe

2024-09-07T01:07:54.788234Z  INFO mistralrs_server: avx: true, neon: false, simd128: false, f16c: true
2024-09-07T01:07:54.788252Z  INFO mistralrs_server: Sampling method: penalties -> temperature -> topk -> topp -> minp -> multinomial
2024-09-07T01:07:54.788263Z  INFO mistralrs_server: Model kind is: normal (no quant, no adapters)
2024-09-07T01:07:54.788776Z  INFO mistralrs_core::pipeline::normal: Loading `tokenizer.json` at `$MODELS/microsoft/Phi-3.5-MoE-instruct`
2024-09-07T01:07:54.788798Z  INFO mistralrs_core::pipeline::normal: Loading `tokenizer.json` locally at `$MODELS/microsoft/Phi-3.5-MoE-instruct/tokenizer.json`
2024-09-07T01:07:54.788801Z  INFO mistralrs_core::pipeline::normal: Loading `config.json` at `$MODELS/microsoft/Phi-3.5-MoE-instruct`
2024-09-07T01:07:54.788809Z  INFO mistralrs_core::pipeline::normal: Loading `config.json` locally at `$MODELS/microsoft/Phi-3.5-MoE-instruct/config.json`
2024-09-07T01:07:54.790795Z  INFO mistralrs_core::pipeline::paths: Found model weight filenames ["model-00001-of-00017.safetensors", "model-00002-of-00017.safetensors", "model-00003-of-00017.safetensors", "model-00004-of-00017.safetensors", "model-00005-of-00017.safetensors", "model-00006-of-00017.safetensors", "model-00007-of-00017.safetensors", "model-00008-of-00017.safetensors", "model-00009-of-00017.safetensors", "model-00010-of-00017.safetensors", "model-00011-of-00017.safetensors", "model-00012-of-00017.safetensors", "model-00013-of-00017.safetensors", "model-00014-of-00017.safetensors", "model-00015-of-00017.safetensors", "model-00016-of-00017.safetensors", "model-00017-of-00017.safetensors"]
2024-09-07T01:07:54.790811Z  INFO mistralrs_core::pipeline::paths: Loading `model-00001-of-00017.safetensors` locally at `$MODELS/microsoft/Phi-3.5-MoE-instruct/model-00001-of-00017.safetensors`
2024-09-07T01:07:54.790820Z  INFO mistralrs_core::pipeline::paths: Loading `model-00002-of-00017.safetensors` locally at `$MODELS/microsoft/Phi-3.5-MoE-instruct/model-00002-of-00017.safetensors`
2024-09-07T01:07:54.790828Z  INFO mistralrs_core::pipeline::paths: Loading `model-00003-of-00017.safetensors` locally at `$MODELS/microsoft/Phi-3.5-MoE-instruct/model-00003-of-00017.safetensors`
2024-09-07T01:07:54.790836Z  INFO mistralrs_core::pipeline::paths: Loading `model-00004-of-00017.safetensors` locally at `$MODELS/microsoft/Phi-3.5-MoE-instruct/model-00004-of-00017.safetensors`
2024-09-07T01:07:54.790844Z  INFO mistralrs_core::pipeline::paths: Loading `model-00005-of-00017.safetensors` locally at `$MODELS/microsoft/Phi-3.5-MoE-instruct/model-00005-of-00017.safetensors`
2024-09-07T01:07:54.790853Z  INFO mistralrs_core::pipeline::paths: Loading `model-00006-of-00017.safetensors` locally at `$MODELS/microsoft/Phi-3.5-MoE-instruct/model-00006-of-00017.safetensors`
2024-09-07T01:07:54.790860Z  INFO mistralrs_core::pipeline::paths: Loading `model-00007-of-00017.safetensors` locally at `$MODELS/microsoft/Phi-3.5-MoE-instruct/model-00007-of-00017.safetensors`
2024-09-07T01:07:54.790868Z  INFO mistralrs_core::pipeline::paths: Loading `model-00008-of-00017.safetensors` locally at `$MODELS/microsoft/Phi-3.5-MoE-instruct/model-00008-of-00017.safetensors`
2024-09-07T01:07:54.790876Z  INFO mistralrs_core::pipeline::paths: Loading `model-00009-of-00017.safetensors` locally at `$MODELS/microsoft/Phi-3.5-MoE-instruct/model-00009-of-00017.safetensors`
2024-09-07T01:07:54.790884Z  INFO mistralrs_core::pipeline::paths: Loading `model-00010-of-00017.safetensors` locally at `$MODELS/microsoft/Phi-3.5-MoE-instruct/model-00010-of-00017.safetensors`
2024-09-07T01:07:54.790892Z  INFO mistralrs_core::pipeline::paths: Loading `model-00011-of-00017.safetensors` locally at `$MODELS/microsoft/Phi-3.5-MoE-instruct/model-00011-of-00017.safetensors`
2024-09-07T01:07:54.790900Z  INFO mistralrs_core::pipeline::paths: Loading `model-00012-of-00017.safetensors` locally at `$MODELS/microsoft/Phi-3.5-MoE-instruct/model-00012-of-00017.safetensors`
2024-09-07T01:07:54.790908Z  INFO mistralrs_core::pipeline::paths: Loading `model-00013-of-00017.safetensors` locally at `$MODELS/microsoft/Phi-3.5-MoE-instruct/model-00013-of-00017.safetensors`
2024-09-07T01:07:54.790916Z  INFO mistralrs_core::pipeline::paths: Loading `model-00014-of-00017.safetensors` locally at `$MODELS/microsoft/Phi-3.5-MoE-instruct/model-00014-of-00017.safetensors`
2024-09-07T01:07:54.790924Z  INFO mistralrs_core::pipeline::paths: Loading `model-00015-of-00017.safetensors` locally at `$MODELS/microsoft/Phi-3.5-MoE-instruct/model-00015-of-00017.safetensors`
2024-09-07T01:07:54.790931Z  INFO mistralrs_core::pipeline::paths: Loading `model-00016-of-00017.safetensors` locally at `$MODELS/microsoft/Phi-3.5-MoE-instruct/model-00016-of-00017.safetensors`
2024-09-07T01:07:54.790939Z  INFO mistralrs_core::pipeline::paths: Loading `model-00017-of-00017.safetensors` locally at `$MODELS/microsoft/Phi-3.5-MoE-instruct/model-00017-of-00017.safetensors`
2024-09-07T01:07:54.791031Z  INFO mistralrs_core::pipeline::normal: Loading `generation_config.json` at `$MODELS/microsoft/Phi-3.5-MoE-instruct`
2024-09-07T01:07:54.791040Z  INFO mistralrs_core::pipeline::normal: Loading `generation_config.json` locally at `$MODELS/microsoft/Phi-3.5-MoE-instruct/generation_config.json`
2024-09-07T01:07:54.791079Z  INFO mistralrs_core::pipeline::normal: Loading `tokenizer_config.json` at `$MODELS/microsoft/Phi-3.5-MoE-instruct`
2024-09-07T01:07:54.791088Z  INFO mistralrs_core::pipeline::normal: Loading `tokenizer_config.json` locally at `$MODELS/microsoft/Phi-3.5-MoE-instruct/tokenizer_config.json`
2024-09-07T01:07:54.791423Z  INFO mistralrs_core::pipeline::normal: Loading model `$MODELS/microsoft/Phi-3.5-MoE-instruct` on cpu.
2024-09-07T01:07:54.793436Z  INFO mistralrs_core::utils::normal: DType selected is F16.
2024-09-07T01:07:54.793461Z  INFO mistralrs_core::pipeline::normal: Model config: Config { vocab_size: 32064, hidden_act: Silu, hidden_size: 4096, intermediate_size: 6400, num_hidden_layers: 32, num_attention_heads: 32, num_key_value_heads: 8, rms_norm_eps: 1e-5, rope_theta: 10000.0, rope_scaling: Some(Classic { short_factor: [1.0, 1.0399999618530271, 1.0399999618530271, 1.0399999618530271, 1.0499999523162842, 1.0499999523162842, 1.0499999523162842, 1.0499999523162842, 1.0499999523162842, 1.0499999523162842, 1.0499999523162842, 1.0499999523162842, 1.0499999523162842, 1.0499999523162842, 1.059999942779541, 1.059999942779541, 1.069999933242798, 1.069999933242798, 1.069999933242798, 1.069999933242798, 1.1399999856948853, 1.159999966621399, 1.159999966621399, 1.159999966621399, 1.159999966621399, 1.1799999475479126, 1.1999999284744265, 1.319999933242798, 1.3399999141693115, 1.3499999046325684, 1.3999998569488523, 1.4799998998641968, 1.4999998807907104, 1.589999794960022, 1.6499998569488523, 1.71999990940094, 1.8999998569488523, 1.9099998474121096, 1.9099998474121096, 1.9899998903274536, 1.9999998807907104, 1.9999998807907104, 2.009999990463257, 2.009999990463257, 2.009999990463257, 2.009999990463257, 2.009999990463257, 2.009999990463257, 2.009999990463257, 2.009999990463257, 2.009999990463257, 2.009999990463257, 2.009999990463257, 2.009999990463257, 2.009999990463257, 2.009999990463257, 2.009999990463257, 2.009999990463257, 2.009999990463257, 2.0999999046325684, 2.319999933242798, 2.419999837875366, 2.5899999141693115, 2.7899999618530273], long_factor: [1.0199999809265137, 1.0299999713897705, 1.0399999618530271, 1.0499999523162842, 1.0499999523162842, 1.0499999523162842, 1.059999942779541, 1.059999942779541, 1.059999942779541, 1.059999942779541, 1.059999942779541, 1.059999942779541, 1.0999999046325684, 1.1799999475479126, 1.1799999475479126, 1.3700000047683716, 1.4899998903274536, 2.109999895095825, 2.8899998664855957, 3.9499998092651367, 4.299999713897705, 6.429999828338623, 8.09000015258789, 10.690000534057615, 12.050000190734863, 18.229999542236328, 18.84000015258789, 19.899999618530273, 21.420000076293945, 26.200000762939453, 34.28000259399414, 34.590003967285156, 38.730003356933594, 40.22000503540039, 42.54000473022461, 44.000003814697266, 47.59000396728515, 54.750003814697266, 56.19000244140625, 57.44000244140625, 57.4900016784668, 61.20000076293945, 61.540000915527344, 61.75, 61.779998779296875, 62.06999969482422, 63.11000061035156, 63.43000030517578, 63.560001373291016, 63.71000289916992, 63.92000198364258, 63.94000244140625, 63.94000244140625, 63.96000289916992, 63.980003356933594, 64.0300064086914, 64.0300064086914, 64.0300064086914, 64.04000854492188, 64.10000610351563, 64.19000244140625, 64.20999908447266, 64.75, 64.95999908447266], scaling_type: Su }), max_position_embeddings: 131072, use_flash_attn: false, sliding_window: Some(131072), original_max_position_embeddings: 4096, quantization_config: None, lm_head_bias: true, attention_bias: true, num_local_experts: 16, router_jitter_noise: 0.01 }
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 118/118 [00:27<00:00, 2.88it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 110/110 [00:33<00:00, 1.04it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 118/118 [00:34<00:00, 2.20it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 118/118 [00:34<00:00, 1.86it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 118/118 [00:35<00:00, 0.66it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 118/118 [00:36<00:00, 1.79it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 118/118 [00:37<00:00, 0.75it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 118/118 [00:40<00:00, 0.65it/s]
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 89/89 [00:41<00:00, 0.55it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 118/118 [00:43<00:00, 0.63it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 118/118 [00:57<00:00, 0.46it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 118/118 [01:02<00:00, 0.86it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 118/118 [01:03<00:00, 1.23it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 112/112 [01:04<00:00, 0.92it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 118/118 [01:04<00:00, 2.03it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 118/118 [01:05<00:00, 1.08it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 112/112 [01:05<00:00, 2.52it/s]
2024-09-07T01:09:13.257129Z  INFO mistralrs_core::pipeline::isq: Applying in-situ quantization into Some(Q6K) to 1665 tensors.
2024-09-07T01:09:13.259066Z  INFO mistralrs_core::pipeline::isq: Applying ISQ on 16 threads.
2024-09-07T01:10:53.974569Z  INFO mistralrs_core::pipeline::isq: Applied in-situ quantization into Some(Q6K) to 1665 tensors out of 1665 total tensors. Took 100.72s
2024-09-07T01:10:54.036317Z  INFO mistralrs_core::pipeline::chat_template: bos_toks = "<s>", eos_toks = "<|endoftext|>", "<|assistant|>", "<|end|>", unk_tok = <unk>
2024-09-07T01:10:54.045908Z  INFO mistralrs_server: Model loaded.
2024-09-07T01:10:54.046519Z  INFO mistralrs_server::interactive_mode: Starting interactive loop with sampling params: SamplingParams { temperature: Some(0.1), top_k: Some(32), top_p: Some(0.1), min_p: Some(0.05), top_n_logprobs: 0, frequency_penalty: Some(0.1), presence_penalty: Some(0.1), stop_toks: None, max_len: Some(4096), logits_bias: None, n_choices: 1, dry_params: Some(DrySamplingParams { sequence_breakers: ["\n", ":", "\"", "*"], multiplier: 0.0, base: 1.75, allowed_length: 2 }) }
> <|system|>You are a helpful assistant.<|end|><|user|>How to explain Internet for a medieval knight?<|end|><|assistant|>
2024-09-07T01:11:26.299096Z ERROR mistralrs_core::engine: prompt step - Model failed with error: Msg("Forward step expected a PagedAttention input metadata. This was not provided, please ensure that the scheduler config is correctly configured for PagedAttention.")
2024-09-07T01:11:26.299328Z ERROR mistralrs_server::interactive_mode: Got a model error: "Forward step expected a PagedAttention input metadata. This was not provided, please ensure that the scheduler config is correctly configured for PagedAttention.", response: ChatCompletionResponse { id: "0", choices: [Choice { finish_reason: "error", index: 0, message: ResponseMessage { content: Some(""), role: "assistant", tool_calls: [] }, logprobs: None }], created: 1725671486, model: "$MODELS/microsoft/Phi-3.5-MoE-instruct", system_fingerprint: "local", object: "chat.completion", usage: Usage { completion_tokens: 0, prompt_tokens: 24, total_tokens: 24, avg_tok_per_sec: 4800.0, avg_prompt_tok_per_sec: inf, avg_compl_tok_per_sec: NaN, total_time_sec: 0.005, total_prompt_time_sec: 0.0, total_completion_time_sec: 0.0 } }
thread 'main' panicked at mistralrs-core/src/lib.rs:262:14:
Cannot block the current thread from within a runtime. This happens because a function attempted to block the current thread while the thread is being used to drive asynchronous tasks.
stack backtrace:
   0: rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::option::expect_failed
   3: tokio::future::block_on::block_on
   4: <mistralrs_core::MistralRs as core::ops::drop::Drop>::drop
   5: alloc::sync::Arc<T,A>::drop_slow
   6: mistralrs_server::interactive_mode::interactive_mode::{{closure}}
   7: mistralrs_server::main::{{closure}}
   8: tokio::runtime::park::CachedParkThread::block_on
   9: mistralrs_server::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

## Latest commit or version
commit 3933afbf8ea0566d64f3fa78a2e4b5cbf0904090 running on a single CPU.

**Comment by EricLBuehler** [comment]
Hi @Remember20240719! Thanks for the issue and the traceback. This was a regression from #753, and I just merged #759 which should fix it. Can you please confirm it works now?

**Comment by Remember20240719** [comment]
@EricLBuehler Yes it's working perfectly! Thank you.
